examples/RunTemporaryCommand/runTempCommand.go

```diff
@@ -0,0 +1,290 @@
+
+	for systemId, result := range systemsFound {
+		if result.Data.ExitCode > 0 || result.Error != "" {
+			dataBuffer += fmt.Sprintf("ID %s: Failed, exit code=%d, error=[%s]\n", systemId, result.Data.ExitCode, result.Error)
+		}
+	}
+
+	if dataBuffer != "" {
+		outputBuffer = headerBuffer + dataBuffer
+	}
+
+	// Output the results for all systems that did not return any result
+	headerBuffer = fmt.Sprintf("\nSystems with no command result\n")
+	headerBuffer += fmt.Sprintf("===============================\n")
+
+	dataBuffer = ""
+
+	for hostName, _ := range systemNameMap {
+		if _, exists := systemsFound[hostName]; !exists {
+			dataBuffer += fmt.Sprintf("ID %s (%s): Received no command result\n", systemNameMap[hostName].Id, hostName)
+		}
+	}
+
+	if dataBuffer != "" {
+		outputBuffer += headerBuffer + dataBuffer
+	}
+
+	return
+}
+
+//
+// This example accepts an API key and the name of the file to execute commands against. You can then
+// specify a regular expression for the JCSystem.Os value of the systems you want the command to run
+// against.
+//
+// It will write the results of the commands to stdout when exitCode != 0
+//
+func main() {
+	apiKey := flag.String("api-key", "none", "Your JumpCloud Administrator API Key")
+	commandFile := flag.String("command-file", "myCommand.txt", "The name of a file containing a command to run")
+	commandType := flag.String("command-type", "windows", "OS type of the command, 'linux', 'windows', or 'mac'")
+	shell := flag.String("shell", "powershell", "Shell (Windows-only) (powershell/cmd)")
+	osType := flag.String("os-type", "Windows.*", "A regular expression to match your systems for OS type")
+	deleteFlag := flag.Bool("delete-after-run", false, "When true, delete commands and results at completion.")
+
+	flag.Parse()
+
+	jc := jcapi.NewJCAPI(*apiKey, URL_BASE)
+
+	// Generate a randomized command name
+	commandName := "CMD " + makeRandomString(COMMAND_NAME_RANDOM_PART)
+
+	// Read the input file
+	commandData, err := readFile(*commandFile, MAX_COMMAND_LEN)
+	if err != nil {
+		log.Fatalf("Could not read command, err='%s'", err.Error())
+	}
+
+	// Make the command object
+	commandObj := makeImmediateCommand(commandName, commandData, *commandType, *shell, jcapi.COMMAND_ROOT_USER)
+
+	//
+	// Get the list of matching servers and add them to the command
+	//
+	systems, err := jc.GetSystems(false)
+	if err != nil {
+		log.Fatalf("Could not get a list of all systems, err='%s'")
+	}
+
+	indices, err := findSystemsByOSType(systems, *osType)
+	if err != nil {
+		log.Fatalf("Could not search a list of systems for OS type matching '%s', err='%s'", *osType, err.Error())
+	}
+
+	if len(indices) == 0 {
+		log.Fatalf("No systems match '%s' on your JumpCloud account\n", *osType)
+	}
+
+	fmt.Printf("Executing Command on the Following Systems\n")
+	fmt.Printf("------------------------------------------\n")
+
+	for _, index := range indices {
+		fmt.Printf("%s\t%s\n", systems[index].Id, systems[index].Hostname)
```

Comment:
While I completely agree from a UX standpoint, I went with Hostname here because that's what ends up on the Command Result as the only way to tie the result back to the system that ran it (I wish we'd included the systems._id, because hostname is not guaranteed to be unique). As something that ends up as example code, I'm a bit leery of confusing the use of Hostname on command results with Displayname.